std/array.d

```diff
@@ -3661,7 +3672,7 @@ unittest
     {
         return [new shared A].inputRangeObject;
     }
-    auto res = foo.array;
+    static assert(__traits(compiles, foo.array));
```

Comment:
>I didn't remove any traits compiles in this PR

https://github.com/dlang/phobos/pull/5676/files#diff-d7e888ff77123ed1925838a861e48e84L1503

>Nesting it with a traits compiles gets rid of the unused variable while preserving functionality

Better to actually test the results than get rid of the variable.